src/main/java/lotto/Rank.java

```diff
@@ -0,0 +1,63 @@
+package lotto;
+
+import java.util.Arrays;
+
+public enum Rank {
+    FIFTH("5등", 3, 5_000),
+    FOURTH("4등", 4, 50_000),
+    THIRD("3등", 5, 1_500_000),
+    SECOND("2등", 5, 30_000_000),
+    FIRST("1등", 6, 2_000_000_000);
+
+    private String name;
+    private int matchCount;
+    private int winningAmount;
+
+    Rank(String name, int matchCount, int winningAmount) {
+        this.name = name;
+        this.matchCount = matchCount;
+        this.winningAmount = winningAmount;
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    public int getMatchCount() {
+        return matchCount;
+    }
+
+    public int getWinningAmount() {
+        return winningAmount;
+    }
+
+    public static Rank valueOf(int matchCount, boolean matchBonus) {
+
+        Rank rank = null;
+        switch (matchCount) {
+            case 3:
+                rank = THIRD;
+                break;
+            case 4:
+                rank = FOURTH;
+                break;
+            case 5:
+                rank = checkBonusBall(matchBonus);
+                break;
+            case 6:
+                rank = FIRST;
+                break;
+        }
+
+        return rank;
+    }
```

Comment:
switch문은 케이스별 처리로직을 부여해야 하기때문에 사실상 if/else문을 사용하는 것과 큰 차이가 없습니다.
아래의 방법에서 bonus 부분 처리만 추가하면 될 것 같습니다 :)

```suggestion
    public static Rank valueOf(int matchCount, boolean matchBonus) {
        return Arrays.stream(values())
                .filter(it -> it.matchCount == matchCount)
                .findFirst()
                .orElseThrow(IllegalArgumentException::new);
    }
```